src/v1/index.ts

```diff
@@ -328,3 +341,114 @@ function decodeBallotVotes(contests: Contests, bits: BitReader): VotesDict {
 
   return votes
 }
+
+interface HMPBBallotPageMetadata {
+  election: Election
+  electionHash: string // a hexadecimal string
+  precinctId: string
+  ballotStyleId: string
+  locales: BallotLocale
+  pageNum: number
+  isLiveMode: boolean
+  isAbsenteeMode: boolean
+  ballotId?: string
+}
+
+export function encodeHMPBBallotPageMetadata({
+  election,
+  electionHash,
+  precinctId,
+  ballotStyleId,
+  locales,
+  pageNum,
+  isLiveMode,
+  isAbsenteeMode,
+  ballotId,
+}: HMPBBallotPageMetadata): Uint8Array {
+  const bits = new BitWriter()
+
+  const precincts = election.precincts
+  const ballotStyles = election.ballotStyles
+
+  const precinctIndex = precincts.findIndex((p) => p.id === precinctId)
+  const ballotStyleIndex = ballotStyles.findIndex(
+    (bs) => bs.id === ballotStyleId
+  )
+  const primaryLocaleIndex = SUPPORTED_LOCALES.indexOf(locales.primary)
+  const secondaryLocaleIndex = locales.secondary
+    ? SUPPORTED_LOCALES.indexOf(locales.secondary)
```

Comment:
This will return -1 if it isn't a known locale. I think it should throw. Same with primary.